pkg/virtualKubelet/reflection/storage/storage_suite_test.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019-2021 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"flag"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+
+	"github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/virtualKubelet/reflection/manager"
+	"github.com/liqotech/liqo/pkg/virtualKubelet/reflection/options"
+)
+
+const (
+	LocalNamespace  = "local-namespace"
+	RemoteNamespace = "remote-namespace"
+
+	LocalClusterID  = "local-cluster"
+	RemoteClusterID = "remote-cluster"
+
+	VirtualStorageClassName    = "liqo"
+	RealRemoteStorageClassName = "other-class"
+
+	VirtualNodeName = "liqo-node"
+	RealNodeName    = "real-node"
+	localPvcName    = "pvc-local"
+	remotePvcName   = "pvc-remote"
+)
+
+var (
+	k8sClient kubernetes.Interface
+
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	reflectorBuilder func(*options.NamespacedOpts) manager.NamespacedReflector
+	factory          informers.SharedInformerFactory
+	reflector        *NamespacedPersistentVolumeClaimReflector
+)
+
+func TestStorage(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Storage Reflection Suite")
+}
+
+var _ = BeforeSuite(func() {
+	klog.SetOutput(GinkgoWriter)
+	flagset := flag.NewFlagSet("klog", flag.PanicOnError)
+	klog.InitFlags(flagset)
+	Expect(flagset.Set("v", "4")).To(Succeed())
+	klog.LogToStderr(false)
+
+	ctx, cancel = context.WithCancel(context.Background())
```

Comment:
Should be moved to the BeforeEach entry (and canceled in an AfterEach), otherwise factories and reflectors are never stopped.